src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -3915,10 +3915,17 @@ void Unit::RemoveNotOwnSingleTargetAuras(uint32 newPhase, bool phaseid)
     for (AuraList::iterator iter = scAuras.begin(); iter != scAuras.end();)
     {
         Aura* aura = *iter;
-        if (aura->GetUnitOwner() != this && !aura->GetUnitOwner()->IsInPhase(newPhase))
+        if (aura->GetUnitOwner() && aura->GetUnitOwner() != this && !aura->GetUnitOwner()->IsInPhase(newPhase))
         {
-            aura->Remove();
-            iter = scAuras.begin();
+            if ((newPhase == 0x0 && !phaseid) ||
+                (!aura->GetEffect(0) || aura->GetEffect(0)->GetAuraType() != SPELL_AURA_CONTROL_VEHICLE) &&
```

Comment:
According to travis-ci you need to add parenthesis around the `&&` expression to clear a logic mismatch:
```cpp
                (!aura->GetEffect(0) || (aura->GetEffect(0)->GetAuraType() != SPELL_AURA_CONTROL_VEHICLE) &&
                (!aura->GetEffect(1)) || aura->GetEffect(1)->GetAuraType() != SPELL_AURA_CONTROL_VEHICLE))
```